### PR TITLE
Added feature to merge sign locks into containers

### PR DESCRIPTION
**Pull request by paulpugh000**
Currently, these changes only allow locked signs to be merged into chests. The ability to lock a chest directly with a command, sign, do not work. Functionality of being able to lock with an IRON_NUGGET or TRIPWIRE_HOOK are only partially working (They only lock the container itself, not the other side of a double chest, if it is one).
The locks inside the container themselves should work the same, although I have not tested this.

### CODE REVIEW

**Comment by paulpugh000** [comment]
All of the methods being in the ChestLockListener is definitely an anti-pattern. I'm looking to possibly move most of the methods into a LockManager file.

File: src/main/java/plugins/nate/smp/commands/SMPCommand.java [hunk 1]
```diff
@@ -106,6 +110,52 @@ public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command
...
+                return true;
+            }
+            Container container = (Container) targetBlock;
+
+            // ChestLockListener chestLockListener = new ChestLockListener();
+            ChestLockListener.lockContainer(player.getUniqueId(), container);
+
+        } else if (args[0].equalsIgnoreCase("unlock")) {
+            if (!(sender instanceof Player)) {
+                sendMessage(sender, "&cOnly players can use this command!");
```
Comment (NRProjects) [inline src/main/java/plugins/nate/smp/commands/SMPCommand.java:132]: Might want to add the ChatUtils.PREFIX variable at the start of the message to keep things uniform.

This also goes for the other instances where its missing

File: src/main/java/plugins/nate/smp/listeners/ChestLockListener.java [hunk 2]
```diff
@@ -14,84 +13,123 @@
...
-
-    // TODO: Prevent multiple people from placing lock signs on a single chest or double chest.
+  
+    // TODO: autolock feature
+    // TODO: admin command to force lock to be of a certain player
 
     @EventHandler(priority = EventPriority.HIGHEST)
     public void onSignChange(SignChangeEvent event) {
-        Player player = event.getPlayer();
+        // TODO: Allow signs to lock chests
```
Comment (NRProjects) [inline src/main/java/plugins/nate/smp/listeners/ChestLockListener.java:37]: Seems like this code may be old and unused

File: src/main/java/plugins/nate/smp/listeners/ChestLockListener.java [hunk 3]
```diff
@@ -231,63 +357,113 @@ private Sign getAttachedSign(Block block) {
...
 
     /**
-     * Returns if player is sign owner or admin
+     * Returns if player is container owner, trusted by owner. <p> 
+     * Optional: allow admins bypass in the check
      */
-    private boolean playerCanInteract(Player player, Sign attachedSign) {
-        return isPlayerOwner(player, attachedSign) || canPlayerBypass(player);
+    private boolean hasAccessToContainerOrLock(Player player, Container container, @Nullable boolean canAdminBypass) {
+        Bukkit.broadcastMessage("if its locked...");
```
Comment (NRProjects) [inline src/main/java/plugins/nate/smp/listeners/ChestLockListener.java:437]: Debug line?

**Review by NRProjects — COMMENTED** [review]
Some small things I noticed glancing over.

**Comment by NRProjects** [comment]
Also very unsure about what your describing in your PR. Could you be more clear?

**Comment by paulpugh000** [comment]
When the branch is fully complete, here are the features it seeks to implement. (_Italicized are yet to be implemented_)
- Locks are now stored within the container itself
 - Sign locks will be deprecated. Making a sign with `[Lock]` locks a container, but it will not stay on the chest.
 - When the owner of a locked chest opens a chest with a locked sign on it, the sign will pop off and the lock gets merged into the chest. 
 - Locking a container by _shift_-right clicking container with an `IRON_NUGGET` or `TRIPWIRE_HOOK`. Useful for mass locking.
- Locks are now supported for Containers (Barrel, Blast Furnace, Brewing Stand, Chest, Dispenser, Dropper, Furnace, Hopper, Smoker), _future plans to add Shulker and Crafter support_
 - _Adding commands_
   -  _`/smp lock <optional name>` locks container player is looking at, and if an admin, can make a lock for someone else_
   - _`/smp unlock` unlocks container player is looking at, with forceable admin unlock_
   - _Possibly `/smp givelock <name>` gives someone a lock_ 
   - _`/smp autolock` locks containers automatically whenever they are placed_
- _Have a visual display on the container to show its locked.

Test conditions will be written after all these features have been implemented.

**Comment by NRProjects** [comment]
> When the branch is fully complete, here are the features it seeks to implement. (_Italicized are yet to be implemented_)
> 
> * Locks are now stored within the container itself
> * Sign locks will be deprecated. Making a sign with `[Lock]` locks a container, but it will not stay on the chest.
> * When the owner of a locked chest opens a chest with a locked sign on it, the sign will pop off and the lock gets merged into the chest.
> * Locking a container by _shift_-right clicking container with an `IRON_NUGGET` or `TRIPWIRE_HOOK`. Useful for mass locking.
> * Locks are now supported for Containers (Barrel, Blast Furnace, Brewing Stand, Chest, Dispenser, Dropper, Furnace, Hopper, Smoker), _future plans to add Shulker and Crafter support_
> * _Adding commands_
>   
>   * _`/smp lock <optional name>` locks container player is looking at, and if an admin, can make a lock for someone else_
>   * _`/smp unlock` unlocks container player is looking at, with forceable admin unlock_
>   * _Possibly `/smp givelock <name>` gives someone a lock_
>   * _`/smp autolock` locks containers automatically whenever they are placed_
> * _Have a visual display on the container to show its locked.
> 
> Test conditions will be written after all these features have been implemented.

Sounds like good features.

This is the only thing I maybe don't want to introduce:
```Have a visual display on the container to show its locked.```

I'd rather have it be a chat message then render something for every chest to show that it's locked. Will be better for players with low-end pcs especially in the case of massage storages placed on the server (megabases)

**Comment by Teelair** [comment]
@NRProjects 
> Will be better for players with low-end pcs especially in the case of massage storages placed on the server (megabases)
I think this is a non concern. There's only 12 additional triangles rendered per chest with this plan. I think there's no way that we run into problems with it. Toggleable would be nice, but performance probably won't be impacted.

**Pull request closed without merging**